src/utils/process_utils/src/process_utils.cpp

```diff
@@ -43,115 +44,47 @@
  * @param args List of arguments for the command.
  * @param output A standard output from the command.
  *
- * @return An exit code from the command.
+ * @return 0 on success.
  */
 int ADUC_LaunchChildProcess(
     const std::string& command,
     std::vector<std::string> args,
     std::string& output) // NOLINT(google-runtime-references)
 {
-#define READ_END 0
-#define WRITE_END 1
+    int ret = 0;
 
-    int filedes[2];
-    const int ret = ADUCPAL_pipe(filedes);
-    if (ret != 0)
-    {
-        Log_Error("Cannot create output and error pipes. %s (errno %d).", strerror(errno), errno);
-        return ret;
-    }
-
-    const int pid = ADUCPAL_fork();
+    std::string redirected_command{ command };
+    redirected_command += " ";
 
-    if (pid == 0)
+    for (const std::string& arg : args)
     {
-        // Running inside child process.
-
-        // Redirect stdout and stderr to WRITE_END
-        ADUCPAL_dup2(filedes[WRITE_END], STDOUT_FILENO);
-        ADUCPAL_dup2(filedes[WRITE_END], STDERR_FILENO);
-
-        ADUCPAL_close(filedes[READ_END]);
-        ADUCPAL_close(filedes[WRITE_END]);
-
-        std::vector<char*> argv;
-        argv.reserve(args.size() + 2);
-        argv.emplace_back(const_cast<char*>(command.c_str())); // NOLINT(cppcoreguidelines-pro-type-const-cast)
-        for (const std::string& arg : args)
-        {
-            argv.emplace_back(const_cast<char*>(arg.c_str())); // NOLINT(cppcoreguidelines-pro-type-const-cast)
-        }
-        argv.emplace_back(nullptr);
-
-        // The exec() functions only return if an error has occurred.
-        // The return value is -1, and errno is set to indicate the error.
-        int status = ADUCPAL_execvp(command.c_str(), &argv[0]);
-
-        fprintf(stderr, "execvp failed, returned %d, error %d\n", status, errno);
-
-        _exit(EXIT_FAILURE);
+        redirected_command += arg;
```

Comment:
not sure but would using std stringstream for redirected_command potentially reduce heap reallocations?

Reply:
possibly.  in addition, a prealloc of the string buffer might be an option.  I'll check